### ISSUE TITLE
Automatically detect additional workspace

### ISSUE THREAD

**Issue by rabbiveesh** [bscan/PerlNavigator#55]
I have a monorepo setup, where I have some code in the root of my repo, and other code in subdirectories.

RIght now, perlnavigator is only picking up that the root of my repo is the `workspaceFolder`, but not in my subdirectories. The subdir is just like any other standard perl dir, with `lib/`, `t/`, `bin/` and the like. It even has a `dist.ini`.

Is there a way to automatically get perlnavigator to pick this up?

**Comment by bscan** [comment]
Yes, this should work. The Navigator is not very workspace centric, but rather path centric. I suspect you just need `lib` on the path. 

Something like `"perlnavigator.includePaths": [ "$workspaceFolder/mySubFolder/lib" ] ` should work. 

In general, most of the path resolution is done by perl itself using `perl -c`, so as long as individual perl files can find their dependencies via `-I` include paths, most of the features should work. Let me know if something doesn't work as expected with this setup.

**Comment by rabbiveesh** [comment]
Well, that's what I'm doing now. So the workspace thing itself isn't quite
supported, then.

**Comment by bscan** [comment]
Correct, the Navigator really isn't workspace oriented, it's oriented around single files. As long as each file does a `use` on any dependency it needs, perl resolves all the files it needs and the Navigator is able to work well. Is there anything workspace-specific that the Navigator isn't currently finding? I suppose the `use` module autocompletion will be missing things from subdirectories.

One thing that could help is automatically finding the root paths and adding them to the include path. For example, if I open `lib/Bar/Baz.pm` which has `package Bar::Baz` at the top, and it tries to `use Bar::Qux`, it's pretty clear that the user needs to include that base `lib` directory to get this to work. Adding it automatically would be a nice convenience feature. This would help for subdirectories in monorepos, and for users who simply open files without any workspace at all. However, this wouldn't work for any other files outside of that `lib` directory. For example, with two different projects as subdirectories with `lib` folders, one project would not be able to import from the other.

**Comment by bhuynhdev** [comment]
Would it be possible to have glob patterns? I have 25 projects which are all is structured as "ProjectDomainName/ProjectName/lib". The workspace folder would be the ProjectDomainName. I wanted to use this string `"$workspaceFolder/*/lib"` so that it can detect a nested `lib` folder
